plugins/inputs/procstat/procstat.go

```diff
@@ -391,6 +457,62 @@ func (p *Procstat) SimpleFindPids(f PIDFinder) ([]PID, map[string]string, error)
 // execCommand is so tests can mock out exec.Command usage.
 var execCommand = exec.Command
 
+type MainStatus struct {
+	Map map[string]string
+}
+
+func BytesToString(b []byte) string {
+	return unsafe.String(&b[0], len(b))
+}
+
+func (p *Procstat) supervisorPIDs() ([]string, map[string]MainStatus, error) {
+	var Services string
+	var servicelist []string
+
+	if p.SupervisorUnit != "" {
+		servicelist = strings.Split(p.SupervisorUnit, ",")
+		Services = strings.Join(servicelist, " ")
+	}
+
+	out, err := execCommand("supervisorctl", "status", Services).Output()
+	if err != nil {
+		if !strings.Contains(err.Error(), "exit status") {
+			return nil, nil, err
+		}
```

Comment:
Does it really make sense to continue here in case of error? If so I would rather like to see a whitelist of errors instead of this very coarse decision.

Reply:
Firstly, thank you for your careful and detailed review, which has been very helpful to me. I have preliminarily reviewed all the suggestions and I believe there are many modification suggestions that need to be adopted. Thank you again.

Secondly, the point you mentioned here is due to the inherent problem with the supervisor. In practice, I found that there were exit errors and command execution was completed, so I continued to execute. You can refer to this
https://github.com/Supervisor/supervisor/issues/1223

Reply:
Currently, only “exit status 3” is retained in the wrong whitelist